src/facade/dragonfly_connection.cc

```diff
@@ -1344,6 +1382,31 @@ Connection::MemoryUsage Connection::GetMemoryUsage() const {
   };
 }
 
+Connection::WeakRef::WeakRef(std::shared_ptr<Connection> ptr, QueueBackpressure* backpressure,
+                             unsigned thread)
+    : ptr_{ptr}, backpressure_{backpressure}, thread_{thread} {
+}
+
+unsigned Connection::WeakRef::Thread() const {
+  return thread_;
+}
+
+Connection* Connection::WeakRef::Get() const {
+  // DCHECK_EQ(ProactorBase::me()->GetPoolIndex(), int(thread_));
```

Comment:
@dranikpg i had to comment this check... otherwise i'm running into the following segfault:

```
F20231203 03:55:48.743674 2493204 dragonfly_connection.cc:1395] Check failed: ProactorBase::me()->GetPoolIndex() == int(thread_) (8 vs. 0) 
*** Check failure stack trace: ***                                                                               
    @     0x559a589b7221  google::LogMessage::Fail()                                                             
    @     0x559a589b7167  google::LogMessage::SendToLog()                                                        
    @     0x559a589b693c  google::LogMessage::Flush()                                                            
    @     0x559a589ba7b4  google::LogMessageFatal::~LogMessageFatal()                                            
    @     0x559a5822077f  facade::Connection::WeakRef::Get()                                                     
    @     0x559a57d4f0d7  dfly::DbSlice::TrackKeys()                                                             
    @     0x559a571398c8  dfly::OpTrackKeys()                                             
    @     0x559a57139c44  _ZZN4dfly7Service15DispatchCommandEN4absl12lts_202308024SpanINS3_IcEEEEPN6facade17ConnectionContextEENKUlPNS_11TransactionEPNS_11EngineShardEE_clESA_SC_
    @     0x559a57165021  _ZZN4dfly11Transaction18ScheduleSingleHopTIRZNS_7Service15DispatchCommandEN4absl12lts_202308024SpanINS5_IcEEEEPN6facade17ConnectionContextEEUlPS0_PNS_11EngineShardEE_EEDTclfp_fpTLDnEEEOT_ENKUlSB_SD_E_clESB_SD_
    @     0x559a571951c6  _ZSt13__invoke_implIN6facade8OpStatusERKZN4dfly11Transaction18ScheduleSingleHopTIRZNS2_7Service15DispatchCommandEN4absl12lts_202308024SpanINS8_IcEEEEPNS0_17ConnectionContextEEUlPS3_PNS2_11EngineShardEE_EEDTclfp_fpTLDnEEEOT_EUlSD_SF_E_JSD_SF_EESJ_St14__invoke_otherOT0_DpOT1_
    @     0x559a57188730  _ZSt8__invokeIRKZN4dfly11Transaction18ScheduleSingleHopTIRZNS0_7Service15DispatchCommandEN4absl12lts_202308024SpanINS6_IcEEEEPN6facade17ConnectionContextEEUlPS1_PNS0_11EngineShardEE_EEDTclfp_fpTLDnEEEOT_EUlSC_SE_E_JSC_SE_EENSt15__invoke_resultISI_JDpT0_EE4typeESJ_DpOSO_
    @     0x559a57180324  _ZSt6invokeIRKZN4dfly11Transaction18ScheduleSingleHopTIRZNS0_7Service15DispatchCommandEN4absl12lts_202308024SpanINS6_IcEEEEPN6facade17ConnectionContextEEUlPS1_PNS0_11EngineShardEE_EEDTclfp_fpTLDnEEEOT_EUlSC_SE_E_JSC_SE_EENSt13invoke_resultISI_JDpT0_EE4typeESJ_DpOSO_
    @     0x559a57177ce5  _ZN4absl12lts_2023080219functional_internal12InvokeObjectIZN4dfly11Transaction18ScheduleSingleHopTIRZNS3_7Service15DispatchCommandENS0_4SpanINS7_IcEEEEPN6facade17ConnectionContextEEUlPS4_PNS3_11EngineShardEE_EEDTclfp_fpTLDnEEEOT_EUlSD_SF_E_NSA_8OpStatusEJSD_SF_EEET0_NS1_7VoidPtrEDpNS1_8ForwardTIT1_E4typeE
    @     0x559a57e9f08b  absl::lts_20230802::FunctionRef<>::operator()()                                                                                                            
    @     0x559a57e71dc3  dfly::Transaction::RunQuickie()                                                                                                                            
    @     0x559a57e744c4  dfly::Transaction::ScheduleUniqueShard()                                                                                                                   
    @     0x559a57e68feb  _ZZN4dfly11Transaction17ScheduleSingleHopEN4absl12lts_2023080211FunctionRefIFN6facade8OpStatusEPS0_PNS_11EngineShardEEEEENKUlvE_clEv
    @     0x559a57e902e5  _ZSt13__invoke_implIvRZN4dfly11Transaction17ScheduleSingleHopEN4absl12lts_2023080211FunctionRefIFN6facade8OpStatusEPS1_PNS0_11EngineShardEEEEEUlvE_JEET_St14__invoke_otherOT0_DpOT1_
    @     0x559a57e8e347  _ZSt10__invoke_rIvRZN4dfly11Transaction17ScheduleSingleHopEN4absl12lts_2023080211FunctionRefIFN6facade8OpStatusEPS1_PNS0_11EngineShardEEEEEUlvE_JEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EESF_E4typeEOSG_DpOSH_
    @     0x559a57e8c7fd  _ZNSt17_Function_handlerIFvvEZN4dfly11Transaction17ScheduleSingleHopEN4absl12lts_2023080211FunctionRefIFN6facade8OpStatusEPS2_PNS1_11EngineShardEEEEEUlvE_E9_M_invokeERKSt9_Any_data
    @     0x559a5728e48f  std::function<>::operator()()                                   
    @     0x559a588f556d  util::fb2::FiberQueue::Run()                                    
    @     0x559a578e710e  _ZZN4dfly11EngineShardC4EPN4util3fb212ProactorBaseEP9mi_heap_sENKUlvE0_clEv                                                                                
    @     0x559a57901b8a  _ZSt13__invoke_implIvZN4dfly11EngineShardC4EPN4util3fb212ProactorBaseEP9mi_heap_sEUlvE0_JEET_St14__invoke_otherOT0_DpOT1_
    @     0x559a578fee41  _ZSt8__invokeIZN4dfly11EngineShardC4EPN4util3fb212ProactorBaseEP9mi_heap_sEUlvE0_JEENSt15__invoke_resultIT_JDpT0_EE4typeEOSA_DpOSB_
    @     0x559a578fc35d  _ZSt12__apply_implIZN4dfly11EngineShardC4EPN4util3fb212ProactorBaseEP9mi_heap_sEUlvE0_St5tupleIJEEJEEDcOT_OT0_St16integer_sequenceImJXspT1_EEE
    @     0x559a578fc3da  _ZSt5applyIZN4dfly11EngineShardC4EPN4util3fb212ProactorBaseEP9mi_heap_sEUlvE0_St5tupleIJEEEDcOT_OT0_
    @     0x559a578fc5ea  _ZN4util3fb26detail15WorkerFiberImplIZN4dfly11EngineShardC4EPNS0_12ProactorBaseEP9mi_heap_sEUlvE0_JEE4run_EON5boost7context5fiberE
    @     0x559a578f97d3  _ZZN4util3fb26detail15WorkerFiberImplIZN4dfly11EngineShardC4EPNS0_12ProactorBaseEP9mi_heap_sEUlvE0_JEEC4IN5boost7context21basic_fixedsize_stackINSD_12stack_traitsEEEEESt17basic_string_viewIcSt11char_traitsIcEERKNSD_12preallocatedEOT_OS9_ENKUlONSD_5fiberEE_clESS_
    @     0x559a579091c2  _ZSt13__invoke_implIN5boost7context5fiberERZN4util3fb26detail15WorkerFiberImplIZN4dfly11EngineShardC4EPNS4_12ProactorBaseEP9mi_heap_sEUlvE0_JEEC4INS1_21basic_fixedsize_stackINS1_12stack_traitsEEEEESt17basic_string_viewIcSt11char_traitsIcEERKNS1_12preallocatedEOT_OSD_EUlOS2_E_JS2_EESQ_St14__invoke_otherOT0_DpOT1_
    @     0x559a57907685  _ZSt8__invokeIRZN4util3fb26detail15WorkerFiberImplIZN4dfly11EngineShardC4EPNS1_12ProactorBaseEP9mi_heap_sEUlvE0_JEEC4IN5boost7context21basic_fixedsize_stackINSE_12stack_traitsEEEEESt17basic_string_viewIcSt11char_traitsIcEERKNSE_12preallocatedEOT_OSA_EUlONSE_5fiberEE_JSS_EENSt15__invoke_resultISP_JDpT0_EE4typeESQ_DpOSX_
    @     0x559a579054e2  _ZSt6invokeIRZN4util3fb26detail15WorkerFiberImplIZN4dfly11EngineShardC4EPNS1_12ProactorBaseEP9mi_heap_sEUlvE0_JEEC4IN5boost7context21basic_fixedsize_stackINSE_12stack_traitsEEEEESt17basic_string_viewIcSt11char_traitsIcEERKNSE_12preallocatedEOT_OSA_EUlONSE_5fiberEE_JSS_EENSt13invoke_resultISP_JDpT0_EE4typeESQ_DpOSX_
*** SIGABRT received at time=1701604548 on cpu 8 ***                                      
PC: @     0x7f72c3dc29fc  (unknown)  pthread_kill                                         
    @     0x559a58a3c3fb         64  absl::lts_20230802::WriteFailureInfo()                                                                                                          
    @     0x559a58a3c655         96  absl::lts_20230802::AbslFailureSignalHandler()                                                                                                  
    @     0x7f72c3d6e520  (unknown)  (unknown)                                            
```